File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
app.service('specificationService', function($http){
    this.findAll=function () {
        return $http.get('../specification/findAll.do');
    }
    this.search=function (page, rows, searchEntity) {
        return $http.post("../specification/search.do?page="+page+"&rows="+rows,searchEntity);
    }
    //增加
    this.add=function (entity) {
        return $http.post("../specification/add.do",entity);
    }
    //修改
    this.update=function (entity) {
        return $http.post("../specification/update.do",entity);
    }
    //修改时查询
    this.findOne=function(id) {
        return $http.get("../specification/findOne.do?id="+id);
    }
    //删除
    this.del=function (ids) {
        return new $http.get("../specification/delete.do?ids="+ids);
    }
    //下拉列表数据
    this.selectSpecificationList=function(){
        return $http.get('../specification/selectSpecificationList.do');
    }
});

File: pinyougou-sellergoods-interface/src/main/java/com/rl/sellergoods/service/SpecificationService.java
package com.rl.sellergoods.service;

import com.rl.pojo.TbSpecification;
import com.rl.pojogroup.Specification;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @Author Ren
 * @Description //TODO 商品规格管理
 * @Date 2018/10/8 10:55
 * @Param
 * @return
 **/
public interface SpecificationService {
    //返回全部列表
    List<TbSpecification> findAll();
    //分页
    PageResult findPage(TbSpecification specification, int page, int rows);
    //增加
    void add(Specification specification);
    //根据id获取实体
    public Specification findOne(Long id);
    //保存修改结果
    void update(Specification specification);
    //批量删除
    void delete(Long[] ids);

    //展示下拉数据
    List<Map> selectSpecificationList();
}
